### DISCUSSION TITLE
ChatGPT 辅助增强

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#4933] [插件说明]
描述
---

> 适配 ChatGPT 页面（包括 ChatGPT Plus 版），并提供一些方便导出到其它生产力工具的辅助性设置。

地址
---

[https://simpread.ksria.cn/plugins/details/8CjJLsNUKU](https://simpread.ksria.cn/plugins/details/8CjJLsNUKU)

适配目的
---

通过此插件适配的模式的效果更加，同时进入简悦的阅读模式后，可导入到任意简悦的支持的生产力工具。

包括：加入稍后读时导入、手动导入均可，甚至还在可在 ChatGPT 页面的基础上进行标注等操作。

导入效果
---

![image](https://user-images.githubusercontent.com/81074/208827015-764be20a-4e7a-48a2-98ef-67f60725c87b.png)

视频
---

进入 ChatGPT 后，导入到 Obsidian

https://user-images.githubusercontent.com/81074/208832848-54cdf031-fe0b-4670-9445-d3a16a3c2f9c.mp4

快捷键
---

`c t`

如何使用
---

1. 更新适配规则

    <img width="924" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/0e94adad-e103-46d3-9582-10069a07abd1">

    关于官方主适配源更多说明 https://github.com/Kenshin/simpread/discussions/2870

2. 将 `chatgpt.com` 加入到延迟加载列表

    <img width="1193" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/a7c8de48-a310-44d8-812d-13fb1d768b39">

选项
---

此插件因为专属给 ChatGPT 做的，所以只能在 https://chatgpt.com/ 下访问，进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1817" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/e37cccc9-27dc-443f-91d0-d01d9b0fe4d8">

功能
---

1. 进入阅读模式自动优化

    <img width="548" alt="image" src="https://user-images.githubusercontent.com/81074/208827306-8780b948-a434-4137-8f03-c8c2194ba133.png">

2. 自定义描述

    <img width="548" alt="image" src="https://user-images.githubusercontent.com/81074/219846509-0960fc9d-df86-434b-b8bb-6597721cb24b.png">

已删除（废弃功能）
---

<details><summary>因 ChatGPT 的功能更新导致一些辅助性功能已过时，因此删除。</summary>

1. 当不存在标题时，将第一个提问设置为阅读模式的标题

    <img width="547" alt="image" src="https://user-images.githubusercontent.com/81074/219845617-e90b63bf-65d6-4f65-b1ea-5576242f9747.png">

2. 适配暗色模式

   > ChatGPT 的暗色模式无法随系统改变。

   <img width="556" alt="image" src="https://user-images.githubusercontent.com/81074/219845711-872d25c7-6f68-475a-a2c1-a428d85994d4.png">
</details>

题图插件
---

如果想要实现截图的效果，需要安装 [题图插件](https://github.com/Kenshin/simpread/discussions/2487) 并按下图所示设置。

```
{"site":"chatgpt.com","pic":"https://source.unsplash.com/1420x708/?nature"}
```

<img width="483" alt="image" src="https://user-images.githubusercontent.com/81074/208828814-2a944cca-50ae-47cc-9939-377608a538b1.png">

更新日志
---

### 1.2.0

- `Optimize` 适配了最新版的 ChatGPT 页面结构。

- `Remove` 删除了一些因新版 ChatGPT 而不再需要的功能，如：暗色模式 & 标题（官方已自行支持）

### 1.1.4

- `Fix` 因为 ChatGPT 页面结构改变而做的新适配。

### 1.1.3

- `Fix` 提问部分无法多行显示的错误

### 1.1.2

- `Fixed` ChatGPT Plus 因为布局改变造成的识别失败

### 1.1.1

- `Optimize` Question 段落样式

- `Optimize` 当深色主题时的 Question 的样式

### 1.1.0

- `Add` 支持 ChatGPT Plus 布局

- `Add` 自动暗色模式

- `Remove` 自定义标题功能（因为 ChatGTP 已经自带此功能）

### 1.0.1

- `Fix` 因 ChatGPT 结构改变造成的适配失败的问题。

### 1.0.0

1. 进入阅读模式自动优化

2. 将第一个提问设置为阅读模式的标题

3.  自定义标题

4. 自定义描述

5. 适配暗色模式

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过[此链接](https://simpread.ksria.cn/plugins/details/8CjJLsNUKU) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

**Comment by amazing-fish** [comment]
能否将默认标题改为openAI现在给出的每个对话的标题？还有我每次会出现openAI的官网图作为标题背景，观感不很好
![image](https://user-images.githubusercontent.com/71763696/211324441-d3b663ed-9206-428d-aa21-0af7a46eb561.png)

**Reply by Kenshin** [reply]
> 能否将默认标题改为openAI现在给出的每个对话的标题？

可以，把下面的选项取消就行。

<img width="555" alt="image" src="https://user-images.githubusercontent.com/81074/211452577-59f26f1b-ac85-4aa2-a18d-4759dd409736.png">

然后默认就是 ChatGPT 的标题了。

<img width="752" alt="image" src="https://user-images.githubusercontent.com/81074/211453118-95af1c29-54c7-4f26-bdc7-10643bedebb9.png">


> 还有我每次会出现openAI的官网图作为标题背景

你可以根据 https://github.com/Kenshin/simpread/discussions/2487#discussioncomment-3330284 修改默认的题图，加入下面的代码

```
{"site":"chat.openai.com","pic":"https://source.unsplash.com/{{win_width}}x708/?nature"}
```

![SCR-20230110-ffn](https://user-images.githubusercontent.com/81074/211453016-a8eb3803-33d8-458c-b88d-3f540f83548c.png)

会变成下图所示

<img width="1212" alt="image" src="https://user-images.githubusercontent.com/81074/211453162-5126f5da-7dfe-491e-b451-195aa9314b10.png">

**Comment by atgnaw** [comment]
跟着示例安装完这个插件，但在chat.openai.com里使用阅读模式，依旧有如下问题。（我安装过官方推荐的notion配置）
![屏幕截图 2023-07-28 231556](https://github.com/Kenshin/simpread/assets/128681516/4f8af4fb-1078-44df-b955-5ba54e3bcbac)
![屏幕截图 2023-07-28 231620](https://github.com/Kenshin/simpread/assets/128681516/e6265993-30aa-43ab-a86f-0de53528db9d)

**Comment by ghost** [comment]
目前ChatGPT支持显示图片了，能否增加上对图片的支持？

**Comment by huhusmang** [comment]
能否支持目前 https://chatgpt.com/ 最新的 UI 呢
![image](https://github.com/Kenshin/simpread/assets/77958410/d0f4a213-fafa-4635-9827-5685872b4591)

**Comment by Kenshin** [comment]
嗨，各位。

@atgnaw @ghost @huhusmang 

ChatGPT 辅助增强插件更新到 **1.2.0**，主要适配了 ChatGPT 最新版界面。


安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过[此链接](https://simpread.ksria.cn/plugins/details/8CjJLsNUKU) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

***

后续有任何问题，请在 https://github.com/Kenshin/simpread/issues/6478 集中讨论。
